addons/stock/models/stock_quant.py

```diff
@@ -285,7 +285,18 @@ def _unlink_zero_quants(self):
         this method is often called in batch and each unlink invalidate
         the cache. We defer the calls to unlink in this method.
         """
-        query = "DELETE FROM stock_quant WHERE round(quantity::numeric, 5) = 0 AND round(reserved_quantity::numeric, 5) = 0;"
+        query = """WITH
+                useless_quants AS (
+                    SELECT quant.id
+                    FROM stock_quant as quant, product_product as product, product_template as template, product_uom as uom
+                    WHERE quant.product_id =  product.id AND
+                    product.product_tmpl_id = template.id AND
+                    template.uom_id = uom.id AND
+                    round(quant.quantity::numeric, (-log(uom.rounding, 10))::integer) = 0 AND
```

Comment:
I would simply take this: https://github.com/odoo/odoo/blob/11.0/addons/product/data/product_data.xml#L49

Reply:
It is better to take the rounding of uom.